src/GTreePluto.cc

```diff
@@ -42,6 +42,37 @@ GTreePluto::ParticleList GTreePluto::GetAllParticles() const
     return list;
 }
 
+PParticle* GTreePluto::GetMCTrue(const int idx) const
+{
+    if (!PlutoMCTrue) {
+        //return nullptr;
+        fprintf(stderr, "ERROR: No Pluto Tree in current file!\n");
+        exit(1);
+    }
+
+    return dynamic_cast<PParticle*>(PlutoMCTrue->At(idx));
+}
+
+TLorentzVector GTreePluto::GetTrueP4(const int idx) const
+{
+    if (!PlutoMCTrue) {
+        fprintf(stderr, "ERROR: No Pluto Tree in current file!\n");
+        exit(1);
+    }
+
+    return dynamic_cast<PParticle*>(PlutoMCTrue->At(idx))->Vect4();
+}
+
+// Assume proton target
+TLorentzVector GTreePluto::GetTrueBeam() const
+{
+    if (!PlutoMCTrue) {
+        fprintf(stderr, "ERROR: No Pluto Tree in current file!\n");
+        exit(1);
+    }
+
+    return dynamic_cast<PParticle*>(PlutoMCTrue->At(0))->Vect4() - TLorentzVector(0., 0., 0., .938272);
```

Comment:
Do NOT assume that particle 0 is always the beam!
This might lead to subtile Bugs! If the user did not save intermediate particles in the pluto data then this is something else!

You have to search the particle list for a particle with ID()==14001 (for proton target) and use that.